src/start_request.cpp

```diff
@@ -54,39 +54,39 @@ StartRequest::RawType StartRequest::toRawType() const
 {
   std::ostringstream os;
 
-  os.write((char*)&crc_, sizeof(crc_));
-  os.write((char*)&seq_number_, sizeof(seq_number_));
-  os.write((char*)&RESERVED_, sizeof(RESERVED_));
-  os.write((char*)&OPCODE_, sizeof(OPCODE_));
+  write<uint32_t>(os, crc_);
```

Comment:
You can skip the template type here. The type is automatically determined by the compiler.